core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunction.java

```diff
@@ -104,19 +90,42 @@ public MeterIdPrefix apply(MeterRegistry registry, RequestLog log) {
                     methodName = requestHeaders.method().name();
                 }
 
-                final List<Tag> tags = new ArrayList<>(4); // method, hostNamePattern, route, status
-                tags.add(Tag.of("method", methodName));
+                tagListBuilder.add(Tag.of("method", methodName));
 
                 if (ctx instanceof ServiceRequestContext) {
                     final ServiceRequestContext sCtx = (ServiceRequestContext) ctx;
-                    tags.add(Tag.of("hostnamePattern", sCtx.virtualHost().hostnamePattern()));
-                    tags.add(Tag.of("route", sCtx.route().meterTag()));
+                    tagListBuilder.add(Tag.of("hostnamePattern", sCtx.virtualHost().hostnamePattern()));
+                    tagListBuilder.add(Tag.of("route", sCtx.route().meterTag()));
                 }
-                return tags;
+            }
+
+            @Override
+            public MeterIdPrefix apply(MeterRegistry registry, RequestLog log) {
+                // method, hostNamePattern, route, status
+                final ImmutableList.Builder<Tag> tagListBuilder = ImmutableList.builderWithExpectedSize(4);
+                buildTags(tagListBuilder, log);
+                appendHttpStatusTag(tagListBuilder, log);
+                return new MeterIdPrefix(name, tagListBuilder.build());
             }
         };
     }
 
+    /**
+     * Append {@link HttpStatus} to {@link Tag}.
+     */
+    static void appendHttpStatusTag(ImmutableList.Builder<Tag> tagListBuilder, RequestLog log) {
```

Comment:
Could you move this to an internal utility class like `armeria.internal.metric.RequestMetricSupport`? It exposes Guava type in the public API.